pkg/components/components.go

```diff
@@ -426,6 +422,21 @@ func GetOperatorCRD() *extv1beta1.CustomResourceDefinition {
 			Subresources: &extv1beta1.CustomResourceSubresources{
 				Status: &extv1beta1.CustomResourceSubresourceStatus{},
 			},
+
+			Validation: &extv1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &extv1beta1.JSONSchemaProps{
+					Properties: map[string]extv1beta1.JSONSchemaProps{
+						"name": extv1beta1.JSONSchemaProps{
+							Type:    "string",
+							Pattern: hcov1alpha1.HyperConvergedName,
+						},
+						"namespace": extv1beta1.JSONSchemaProps{
+							Type:    "string",
+							Pattern: namespace,
+						},
```

Comment:
This validation hierarchy is not correct. It should be:
validation.openAPIV3Schema.properties.**metadata.properties**
see:
https://github.com/kubernetes-sigs/kubebuilder/issues/1074
I checked both yamls and the latter works, the former doesn't.

Reply:
good catch